build.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use protobuf_build::*;
+use std::fs::{read_dir, remove_file, File};
+use std::io::Write;
+
+#[allow(unused_must_use)]
+fn main() {
+    // This build script creates files in the `src` directory. Since that is
+    // outside Cargo's OUT_DIR it will cause an error when this crate is used
+    // as a dependency. Therefore, the user must opt-in to regenerating the
+    // Rust files.
+    if !cfg!(feature = "regenerate") {
+        println!("cargo:rerun-if-changed=build.rs");
+        return;
+    }
+
+    check_protoc_version();
+
+    let file_names: Vec<_> = read_dir("proto")
+        .expect("Couldn't read proto directory")
+        .filter_map(|e| {
+            let e = e.expect("Couldn't list file");
+            if e.file_type().expect("File broken").is_dir() {
+                None
+            } else {
+                Some(format!("proto/{}", e.file_name().to_string_lossy()))
+            }
+        })
+        .collect();
+
+    for f in &file_names {
+        println!("cargo:rerun-if-changed={}", f);
+    }
+
+    // Delete the previously-generated wrapper, we're ok if the file does not exist
+    remove_file("rsprost/protobuf_compat.rs");
+
+    // Generate Prost files.
+    generate_prost_files(&file_names, "src/rsprost");
```

Comment:
Could you change `rsprost` to just `prost`?

Reply:
I didn't do that because there's also an external crate called prost. Do you think it's a problem?